pyinteraph/centrality_analysis.py

```diff
@@ -0,0 +1,604 @@
+#                                                                  normalized = kwargs['norm'])
+#     return centrality_dict
+
+def get_dict_with_group_val(G, node_list, value):
+    """
+    Takes in a graph, list of nodes and a group value. Returns a dict
+    containing each node in the graph. If the node is in the list, its
+    value is the group value or else it is 0.
+    """
+
+    node_dict = {n : (value if n in node_list else 0) for n in G.nodes()}
+    return node_dict
+
+
+def get_group_betweenness_cent(G, **kwargs):
+    """Returns a dictionary of group betweeness centrality values."""
+
+    centrality_val = nxc.group_betweenness_centrality(G = G,
+                                                      C = kwargs['node_list'],
+                                                      normalized = kwargs['norm'],
+                                                      weight = kwargs['weight_name'])
+    centrality_dict = get_dict_with_group_val(G, kwargs['node_list'], centrality_val)
+    return centrality_dict
+
+def get_group_closeness_cent(G, **kwargs):
+    """Returns a dictionary of group closeness centrality values."""
+
+    centrality_val = nxc.group_closeness_centrality(G = G,
+                                                    S = kwargs['node_list'],
+                                                    weight = kwargs['weight_name'])
+    centrality_dict = get_dict_with_group_val(G, kwargs['node_list'], centrality_val)
+    return centrality_dict
+
+def get_edge_betweenness_cent(G, **kwargs):
+    """Returns a dictionary of edge betweenness centrality values."""
+    
+    centrality_dict = nxc.edge_betweenness_centrality(G = G,
+                                                      normalized = kwargs['norm'],
+                                                      weight = kwargs['weight_name'])
+    return centrality_dict
+
+def get_centrality_dict(cent_list, function_map, graph, **kwargs):
+    """Returns two dictionaries. For the first dictionary, the key is the 
+    name of a node centrality measure and the value is a dictionary of 
+    centrality values for each node. (Also includes group centralities)
+    e.g. {degree: {A: 0.1, B:0.7, ...}, betweenness: {...}, ...}
+    For the second dictionary, the key is the name of an edge centrality
+    measure and the value is a dictionary of centrality values for each
+    edge, similar to the first dictionary.
+    """
+
+    node_dict = {}
+    edge_dict = {}
+    for name in cent_list:
+        # Get dictionary using the function map
+        cent_dict = function_map[name](G = graph, **kwargs)
+        # Add edge centralities to edge dict
+        if "edge" in name:
+            edge_dict[name] = cent_dict
+        # Add node centralities to node dict
+        else:
+            node_dict[name] = cent_dict
+    return node_dict, edge_dict
+
+def write_table(fname, centrality_dict, identifiers, sort_by):
```

Comment:
We can think of having the table by `pandas` since it is way tidier and it is the standard for data frames, but that would add a dependency to PyInteraph. What do you think @mtiberti and @elenapapaleo?